backend/src/nodes/nodes/image_utility/convolve.py

```diff
@@ -0,0 +1,87 @@
+from __future__ import annotations
+
+import numpy as np
+
+from . import category as ImageUtilityCategory
+from ...node_base import NodeBase
+from ...node_factory import NodeFactory
+from ...properties.inputs import ImageInput, NumberInput, TextInput
+from ...properties.outputs import ImageOutput
+from ...properties import expression
+
+
+@NodeFactory.register("chainner:image:image_convolve")
+class ImageConvolveNode(NodeBase):
+    def __init__(self):
+        super().__init__()
+        self.description = "Convolves input image with input kernel (kernel values separated by commas)"
+        self.inputs = [
+            ImageInput(),
+            TextInput("Kernel String"),
+            NumberInput("Kernel Width/Height", minimum=0, default=3),
+            NumberInput("Padding", minimum=0, default=0),
+            NumberInput("Strides", minimum=0, default=1),
+        ]
+        self.outputs = [
+            ImageOutput(image_type=expression.Image(size_as="Input0"), channels=1)
+        ]
+        self.category = ImageUtilityCategory
+        self.name = "Convolve"
+        self.icon = "MdAutoFixHigh"
+        self.sub = "Miscellaneous"
+
+    def run(
+        self,
+        img: np.ndarray,
+        kernel_in: str,
+        kernel_dim: int,
+        padding: int,
+        strides: int,
+    ) -> np.ndarray:
+
+        kernel = np.array([float(d) for d in kernel_in.split(",")])
+        kernel = kernel.reshape(kernel_dim, kernel_dim)
+
+        # Grayscale image if it is not already
+        if len(img.shape) != 2:
+            img = img[:, :, 0]
+
+        # Thanks Samrat Sahoo on Medium for the convolution code
+        kernel = np.flipud(np.fliplr(kernel))
+
+        xKernShape = kernel.shape[0]
+        yKernShape = kernel.shape[1]
+        xImgShape = img.shape[0]
+        yImgShape = img.shape[1]
+
+        xOutput = int(((xImgShape - xKernShape + 2 * padding) / strides) + 1)
+        yOutput = int(((yImgShape - yKernShape + 2 * padding) / strides) + 1)
+        output = np.zeros((xOutput, yOutput))
+
+        if padding != 0:
+            imagePadded = np.zeros(
+                (img.shape[0] + padding * 2, img.shape[1] + padding * 2)
+            )
+            imagePadded[
+                int(padding) : int(-1 * padding), int(padding) : int(-1 * padding)
+            ] = img
+        else:
+            imagePadded = img
+
+        for y in range(img.shape[1]):
+            if y > img.shape[1] - yKernShape:
+                break
+            if y % strides == 0:
+                for x in range(img.shape[0]):
```

Comment:
Implementing the convolution yourself isn't ideal for one reasons: Python is slow as heck. Chainner is used on images that with millions of pixels, and pixel-wise operations in Python code are just too slow for that. (There are also more efficient algorithms for convolution based on DTF, but they only really matter for larger kernel sizes starting at around 11x11.)

Please use [`cv2.filter2D`](https://docs.opencv.org/3.4/d4/d86/group__imgproc__filter.html#ga27c049795ce870216ddfb366086b5a04) instead. This function does the convolution for you and is implemented in highly optimized C code. This function is really fast (even for ridiculously large kernel sizes). 
There are good examples for its usage online, but we also use this function throughout chainner, so you can also find a few examples in our code base.

Reply:
Oh man, I can't believe I missed this function. I did realise that this 2D convolution was extremely slow on large images but couldn't find a solution somehow. I'll implement this as well.